Content.Server/GameObjects/Components/Kitchen/KitchenMicrowaveComponent.cs

```diff
@@ -0,0 +1,416 @@
+            UpdateUserInterface();
+            return true;
+
+        }
+
+        //This is required. It's 'cook'.
+        private void wzhzhzh()
+        {
+            if (!HasContents)
+            {
+                return;
+            }
+            
+            _busy = true;
+            // Convert storage into Dictionary of ingredients
+            _solids.Clear();
+            foreach(var item in _storage.ContainedEntities)
+            {
+                if(_solids.ContainsKey(item.Prototype.ID))
+                {
+                    _solids[item.Prototype.ID]++;
+                }
+                else
+                {
+                    _solids.Add(item.Prototype.ID, 1);
+                }
+            }
+
+            // Check recipes
+            FoodRecipePrototype recipeToCook = null;
+            foreach(var r in _recipeManager.Recipes)
+            {
+                if (!CanSatisfyRecipe(r))
+                {
+                    continue;
+                }
+
+                recipeToCook = r;
+            }
+
+            var goodMeal = (recipeToCook != null)
+                           &&
+                           (_currentCookTimerTime == (uint)recipeToCook.CookTime) ? true : false;
+
+            SetAppearance(MicrowaveVisualState.Cooking);
+            _audioSystem.Play(_startCookingSound);
+            Timer.Spawn((int)(_currentCookTimerTime * _cookTimeMultiplier), () =>
+            {
+
+                if (goodMeal)
+                {
+                    SubtractContents(recipeToCook);
+                }
+                else
+                {
+                    VaporizeReagents();
+                    VaporizeSolids();
+                }
+
+                var entityToSpawn = goodMeal ? recipeToCook.Result : _badRecipeName;
+                _entityManager.SpawnEntity(entityToSpawn, Owner.Transform.GridPosition);
+                _audioSystem.Play(_cookingCompleteSound);
+                SetAppearance(MicrowaveVisualState.Idle);
+                _busy = false;
+            });
+            UpdateUserInterface();
+            return;
+        }
+
+        private void VaporizeReagents()
+        {
+            _solution.RemoveAllSolution();
+        }
+
+        private void VaporizeReagentWithIndex()
+        {
+            
+        }
+
+        private void VaporizeSolids()
+        {
+            for(var i = _storage.ContainedEntities.Count-1; i>=0; i--)
+            {
+                var item = _storage.ContainedEntities.ElementAt(i);
+                _storage.Remove(item);
+                item.Delete();
+            }
+            _solids.Clear();
+        }
+
+        private void EjectSolids()
+        {
+
+            for(var i = _storage.ContainedEntities.Count-1; i>=0; i--)
+            {
+                _storage.Remove(_storage.ContainedEntities.ElementAt(i));
+            }
+
+            _solids.Clear();
+        }
+
+        private void EjectSolidWithIndex(EntityUid entityID)
```

Comment:
You should verify that this entity ID actually exists so that a malicious/desynced client couldn't throw an exception.